marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -0,0 +1,314 @@
+  Hedgehog.assert (equivalentLists computedEvent postGenesisEvents)
+
+-- Insert Utxo events in storage, and retreive the events
+--
+utxoStorageTest :: Property
+utxoStorageTest = property $ do
+  events <- forAll genUtxoEvents
+  (storedEvents :: [StorableEvent Utxo.UtxoHandle]) <-
+    (liftIO . Utxo.open ":memory:") (Utxo.Depth 10)
+     >>= liftIO . Storable.insertMany events
+     >>= liftIO . Storable.getEvents
+  Hedgehog.assert (equivalentLists storedEvents events)
+
+-- Insert Utxo events in storage, and retrieve the events by address
+--
+utxoInsertAndQueryTest :: Property
+utxoInsertAndQueryTest = property $ do
+  events <- forAll genUtxoEvents
+  depth <- forAll $ Gen.int (Range.linear 1 5)
+  indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
+             >>= liftIO . Storable.insertMany events
+  let
+    qs :: [StorableQuery Utxo.UtxoHandle]
+    qs = fmap (Utxo.UtxoAddress . Utxo._address) . concatMap (Set.toList . Utxo.ueUtxos) $ events
+  results <- liftIO . traverse (Storable.query Storable.QEverything indexer) $ qs
+  let rows = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
+  computedEvent <-
+    liftIO . Utxo.rowsToEvents (Utxo.getTxIns (getConn indexer) ) $ rows
+  Hedgehog.assert (equivalentLists
+                   computedEvent
+                   (filter (\e -> Utxo.ueChainPoint e /= C.ChainPointAtGenesis) events) )
+
+-- Insert Utxo events in storage, and retreive the events by address and query interval
+--
+utxoQueryIntervalTest :: Property
+utxoQueryIntervalTest = property $ do
+  event0 <- forAll $ genEventAtChainPoint C.ChainPointAtGenesis
+  event1 <- forAll $ genEventAtChainPoint (head chainpoints)
+  event2 <- forAll $ genEventAtChainPoint (chainpoints !! 1)
+  event3 <- forAll $ genEventAtChainPoint (chainpoints !! 2)
+  let events = [event0, event1, event2, event3]
+  indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth 2)
+             >>= liftIO . Storable.insertMany [event0, event1, event2, event3]
+  let
+    qs :: [StorableQuery Utxo.UtxoHandle]
+    qs = fmap (Utxo.UtxoAddress . Utxo._address) . concatMap (Set.toList . Utxo.ueUtxos) $ events
+  results <- liftIO . traverse (Storable.query (Storable.QInterval (head chainpoints)(chainpoints !! 1)) indexer) $ qs
+  let rows = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
+  computedEvent <-
+    liftIO . Utxo.rowsToEvents (Utxo.getTxIns (getConn indexer) ) $ rows
+  Hedgehog.assert (equivalentLists computedEvent [event0, event1])
+
+-- TargetAddresses are the addresses in UTXO that we filter for.
+-- Puporse of this test is to filter out utxos that have a different address than those in the TargetAddress list.
+eventsAtAddressTest :: Property
+eventsAtAddressTest = property $ do
+    event <- head <$> forAll genUtxoEvents
+    let (addresses :: [StorableQuery Utxo.UtxoHandle]) =
+          ( traversed %~ Utxo.UtxoAddress)
+          . nub
+          . toListOf (folded . Utxo.address)
+          . Utxo.ueUtxos
+          $ event
```

Comment:
I know the above is not changed in this PR, but the above could be written as `map (Utxo.UtxoAddress . Utxo._address) $ Set.toList $ Utxo.ueUtxos event` which would be much clearer.

Reply:
Thanks :)